src/main/scala/uk/gov/hmrc/SbtArtifactory.scala

```diff
@@ -183,17 +178,18 @@ object SbtArtifactory extends sbt.AutoPlugin{
     } else {
       // Determine whether we're pointing at labs or live artifactory, and resolve to use the correct releases repo
       (for {
-        artifactoryUri <- artifactoryUriToResolve
-        labNum <- artifactoryLabsPattern.findFirstMatchIn(artifactoryUri).map(_.group(1))
-      } yield s"releases-lab$labNum").getOrElse("releases")
+         artifactoryUri <- artifactoryUriToResolve
+         labNum         <- artifactoryLabsPattern.findFirstMatchIn(artifactoryUri).map(_.group(1))
+       } yield s"releases-lab$labNum"
+      ).getOrElse("releases")
     }
 
   private def artifactoryConnector(repositoryName: String) = new ArtifactoryConnector(
     DispatchCrossSupport.http,
     directCredentials(
-      getOrError(maybeArtifactoryUri, artifactoryUriEnvKey),
-      getOrError(maybeArtifactoryUsername, artifactoryUsernameEnvKey),
-      getOrError(maybeArtifactoryPassword, artifactoryPasswordEnvKey)
+      uri      = getOrError(maybeArtifactoryUri, artifactoryUriEnvKey),
```

Comment:
Can we refactor so we can use `maybeArtifactoryCredentials` here now? 

The difference seems to be that here we want to know which environment key wasn't set. With the logging you've added, we'll know that one of these wasn't set, but not which one.